core/object/message_queue.cpp

```diff
@@ -116,10 +128,18 @@ Error MessageQueue::push_callablep(const Callable &p_callable, const Variant **p
 
 	int room_needed = sizeof(Message) + sizeof(Variant) * p_argcount;
 
-	if ((buffer_end + room_needed) >= buffer_size) {
-		print_line("Failed method: " + p_callable);
-		statistics();
-		ERR_FAIL_V_MSG(ERR_OUT_OF_MEMORY, "Message queue out of memory. Try increasing 'memory/limits/message_queue/max_size_kb' in project settings.");
+	if ((buffer_end + room_needed) >= buffer.size()) {
+		if ((buffer_end + room_needed) >= max_allowed_buffer_size) {
+			String type;
+			if (ObjectDB::get_instance(p_callable.get_object_id())) {
+				type = ObjectDB::get_instance(p_callable.get_object_id())->get_class();
+			}
+			print_line("Failed method: " + p_callable);
+			statistics();
+			ERR_FAIL_V_MSG(ERR_OUT_OF_MEMORY, "Message queue out of memory. Try increasing 'memory/limits/message_queue/max_size_kb' in project settings.");
```

Comment:
```suggestion
			ERR_FAIL_V_MSG(ERR_OUT_OF_MEMORY, "Message queue out of memory. Try increasing 'memory/limits/message_queue/max_size_mb' in project settings.");
```

Reply:
Should also be renamed in `ProjectSettings.xml`, keep the description (or improving it).